docs/docs/installation/docker.mdx

```diff
@@ -0,0 +1,51 @@
+# Docker Images and Tags
+
+The Apache Superset community extensively uses Docker for development, release,
+and productionizing Superset. This page details our Docker builds and tag naming
+schemes to help users navigate our offerings.
+
+Images are built and pushed to the [Superset Docker Hub repository](
+https://hub.docker.com/r/apache/superset). Different sets of images are created for:
+- Published releases, with tags like `3.0.0` and the `latest` tag.
+- Pull request iterations, each identified by tags starting with a SHA like
+  `8a2f7d378ab13c156fa183d9284b607ed69f5ecc`, and `pr-3454`, referencing the pull
+  request ID.
+- Merges to the main branch (`master`), resulting in new SHAs, with tags
+  prefixed with `master` for the latest `master` version.
+
+Each code version has multiple builds for different purposes, identified by suffixes:
+- **Build Preset:** We offer various images for different needs:
+  - **Lean:** The default Docker image, including both frontend and backend. Tags
+    without a build_preset are lean builds, e.g., `latest`.
+  - **Dev:** For development, with a headless browser and root access.
+  - **Py310:** Similar to lean but with Python 3.10.
+  - **CI:** For certain CI workloads.
+  - **WebSocket:** For Superset clusters supporting advanced features.
+  - **Dockerize:** Used by Helm.
+- **Platform:** We build for `linux/arm64` and `linux/amd64`. The `-arm` suffix
+  indicates ARM builds (e.g., `latest-arm`), while tags without a suffix are for
+  AMD (e.g., `latest`).
```

Comment:
I see people asking about ARM builds for certain Mac chipsets.  If we know of any that require ARM, maybe we state that here.

Reply:
Actually when I first got into this, I assumed we introduced `linux/arm64` for Mac workflows, but since then realized there might be a `linux/arm64/v8` for Macs, or at least my M2 laptop seemed to think it needed that. Unclear if Mac M1s use `linux/arm64` or something else. It also brings up the question of if it's not for Macs, how common is `linux/arm64` and should we actually build EVERY COMMIT for EVERY PR on that!?

I personally don't like docker on my dev env much unless I absolutely have to.

Ok, I did a bit of research and while by default my MAC wants that `linux/arm64/v8`, it looks like it'll work with `linux/arm64`, but requires something like this -> 
```
export DOCKER_DEFAULT_PLATFORM=linux/amd64
```

I'll add a note in the docs, though I really don't have much context here. Maybe a v8 build would perform better? If we build on v8, is that compatible with say AWS-hosted arm offering? Does it matter? idk...

Reply:
@sfirke I added a section bellow "On ARM builds and working with Apple silicon"

Reply:
Awesome, thanks for doing the legwork and experimentation with Mac & ARM.  People are often asking about it.  I committed two trivial copyediting changes to that section.